src/vaadin-context-menu.html

```diff
@@ -650,20 +650,39 @@
           }
 
           if (!alignedToParent) {
-            // Sub-menu is displayed in the right side of root menu
-            if ((x < wdthVport / 2 || x < xMax) && !parent) {
-              style.left = x + 'px';
-            } else if ((parent && (wdthVport - parentContentRect.width - parentContentRect.left
-              >= parentContentRect.width))) { // Sub-menu is displayed in the right side of root menu If it is nested menu
-              style.left = parentContentRect.left + parentContentRect.width + 'px';
-            } else if (parent) { // Sub-menu is displayed in the left side of root menu If it is nested menu
-              style.right = 'auto';
-              style.left = Math.max(overlay.getBoundingClientRect().left,
-                parentContentRect.left - overlay.getBoundingClientRect().width) + 'px';
-              overlay.setAttribute('right-aligned', '');
-            } else { // Sub-menu is displayed in the left side of root menu
-              style.right = Math.max(0, (wdthVport - x)) + 'px';
-              overlay.setAttribute('right-aligned', '');
+            if (!this.__isRTL()) {
+              // Sub-menu is displayed in the right side of root menu
+              if ((x < wdthVport / 2 || x < xMax) && !parent) {
+                style.left = x + 'px';
+              } else if ((parent && (wdthVport - parentContentRect.width - parentContentRect.left
+                >= parentContentRect.width))) { // Sub-menu is displayed in the right side of root menu If it is nested menu
+                style.left = parentContentRect.left + parentContentRect.width + 'px';
+              } else if (parent) { // Sub-menu is displayed in the left side of root menu If it is nested menu
+                style.right = 'auto';
+                style.left = Math.max(overlay.getBoundingClientRect().left,
+                  parentContentRect.left - overlay.getBoundingClientRect().width) + 'px';
+                overlay.setAttribute('right-aligned', '');
+              } else { // Sub-menu is displayed in the left side of root menu
+                style.right = Math.max(0, (wdthVport - x)) + 'px';
+                overlay.setAttribute('right-aligned', '');
+              }
+            } else {
+              // Sub-menu is displayed in the left side of root menu
+              if ((x > wdthVport / 2 || x > xMin) && !parent) {
+                style.right = Math.max(0, (wdthVport - x)) + 'px';
+              } else if (parent && (parentContentRect.left >= parentContentRect.width)) {
+                // Sub-menu is displayed in the left side of root menu If it is nested menu
+                style.right = (wdthVport - parentContentRect.right + parentContentRect.width) + 'px';
+              } else if (parent) { // Sub-menu is displayed in the right side of root menu If it is nested menu
```

Comment:
A bit confused over here: if `sub-menu is displayed in the right side of root menu` why we are setting `right-aligned` below. 🤔It means that it's the same as in `ltr` above `displayed in the left side`. Should it work this way?

Reply:
I think the name `right-aligned` became a bit misleading now, but I didn't want to touch it, as it could somehow be a breaking change. But basically, from what I could understand, that attribute means that the context-menu will open a sub-menu towards the opposite direction of the "natural flow". So, in the sense of the `rtl`, `right-aligned` means `left-aligned` (😅).

Reply:
We can actually consider renaming that one to smth like `end-aligned` | `aligned-end`. Not sure if it is a blocker, but we can at least ask from the team :)